src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,444 @@
+      lr_fs->destination.home, "", &dirlist_dest, &num_dest_files));
+
+    switch (state) {
+    case WTI_LIVE_RESTORE_STATE_NONE:
+        /*
+         * Ideally we'd prevent live restore from starting when there are any files already present
+         * in the destination, but we can't control for everything that the user might put into the
+         * folder. Instead only check for WiredTiger files.
+         */
+        for (uint32_t i = 0; i < num_dest_files; ++i) {
+            if (WT_PREFIX_MATCH(dirlist_dest[i], WT_WIREDTIGER) ||
+              WT_SUFFIX_MATCH(dirlist_dest[i], ".wt"))
+                WT_ERR_MSG(session, EINVAL,
+                  "Attempting to begin a live restore on a directory that already contains "
+                  "WiredTiger files '%s'! It's possible this file will be overwritten.",
+                  dirlist_dest[i]);
+        }
+        break;
+    case WTI_LIVE_RESTORE_STATE_LOG_COPY:
+        for (uint32_t i = 0; i < num_dest_files; ++i) {
+            if (!WT_SUFFIX_MATCH(dirlist_dest[i], ".log") &&
+              strcmp(dirlist_dest[i], WTI_LIVE_RESTORE_STATE_FILE) != 0)
+                WT_ERR_MSG(session, EINVAL,
+                  "Live restore state is in log copy phase but the destination contains files "
+                  "other than logs or the state file: %s",
+                  dirlist_dest[i]);
+        }
+        break;
+    case WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION:
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        /* There's no invalid state to check in these cases. */
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        for (uint32_t i = 0; i < num_dest_files; ++i) {
+            if (WT_SUFFIX_MATCH(dirlist_dest[i], WTI_LIVE_RESTORE_STOP_FILE_SUFFIX))
+                WT_ERR_MSG(session, EINVAL,
+                  "Live restore is complete but live restore stop file '%s' still exists!",
+                  dirlist_dest[i]);
+        }
+        break;
+    }
+
+err:
+    WT_TRET(lr_fs->os_file_system->fs_directory_list_free(
+      lr_fs->os_file_system, (WT_SESSION *)session, dirlist_source, num_source_files));
+
+    WT_TRET(lr_fs->os_file_system->fs_directory_list_free(
+      lr_fs->os_file_system, (WT_SESSION *)session, dirlist_dest, num_dest_files));
+
+    return (ret);
+}
+
+/*
+ * __wt_live_restore_init_stats --
+ *     Initialize the live restore stats.
+ */
+void
+__wt_live_restore_init_stats(WT_SESSION_IMPL *session)
+{
+    if (F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_FS)) {
+        /*
+         * The live restore external state is known on initialization, but at that time the stat
+         * server hasn't begun so we can't actually set the state. This must be called after the
+         * stat server starts.
+         */
+        WTI_LIVE_RESTORE_FS *lr_fs = ((WTI_LIVE_RESTORE_FS *)S2C(session)->file_system);
+        WTI_LIVE_RESTORE_STATE state = lr_fs->state;
```

Comment:
A direct read of state without calling get_state? Should probably either call it or check that the lock is taken.

Reply:
https://github.com/wiredtiger/wiredtiger/pull/11563/commits/354f8b06dd66052168ce3dd657584123bce7253e